src/Repositories/UserRepository.cs

```diff
@@ -1,14 +1,44 @@
+using Coffee_Shop_App.src.Databases;
 using Coffee_Shop_App.src.Entities;
+using Coffee_Shop_Appe.src.Abstractions;
 
-namespace Coffee_Shop_App.src.Repositories
+namespace Coffee_Shop_App;
+
+public class UserRepository : IUserRepository
 {
 
-    public class UserRepository 
+    private List<User>? _users; //this to get users info as a list
+    public UserRepository() //constructor to get users data from DB
     {
-       public string SayHi(){
 
-        return "testing";
-       }
+        _users = new DatabaseContext().users; // new obj database to get users' list
+
     }
 
+    public List<User> FindAll()
+    {
+
+        return _users;
+
+    }
+    public User CreateOne(User user)
+    {
+        _users?.Add(user); // temporary saving data
+        return user; //to show new user only
+    }
+
+
+    public User? findOne(string userId) //targeting ids in params
+    {
+        User? user = _users?.FirstOrDefault( user => user.Id == userId); //lambda expression to compare Ids
+        
+        return user; //to get the desired user
+    }
+
+    public User? findOneByEmail(User userEmail)
```

Comment:
here the data type should be `string` because you are passing the email as a `string` not as a `User`